ext/MitsubaLoader/CMitsubaLoader.h

```diff
@@ -16,6 +16,304 @@ namespace ext
+		//ior[4..6] is column 1 (imaginary part of IoR)
+		float ior[8];
+		//same as SAllDielectric::specReflectance
+		float specReflectance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec2 int_ext_ior;
+		vec2 alpha_u;
+		vec2 alpha_v;
+		vec3 specReflectance;
+		vec3 specTransmittance;
+		vec3 diffReflectance;
+	};
+	*/
+	struct alignas(16) SAllPlastic
+	{
+		uint32_t bitfields;
+		float dummy0;
+		//same as SAllDielectric::int_ext_ior
+		float int_ext_ior[2];
+		//same as for SAllDielectric::alpha_u,alpha_v
+		float alpha_u[2];
+		float alpha_v[2];
+		//same as SAllDielectric::specReflectance
+		float specReflectance[3];
+		float dummy1;
+		//same as SAllDielectric::specTransmittance
+		float specTransmittance[3];
+		float dummy2;
+		//if flag decides to use diffuse refl texture, diffReflectance[0..1] is bindless texture ID
+		//otherwise diffReflectance[0..2] is constant diffuse refl
+		float diffReflectance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		float thickness;
+		vec2 int_ext_ior;
+		vec2 alpha_u;
+		vec2 alpha_v;
+		vec3 specReflectance;
+		vec3 specTransmittance;
+		vec3 sigmaA;
+	};
+	*/
+	struct alignas(16) SAllCoating
+	{
+		//child index (into BSDF buffer) is stored on 15 highest bits
+		uint32_t bitfields;
+		float thickness;
+		//same as SAllDielectric::int_ext_ior
+		float int_ext_ior[2];
+		//same as for SAllDielectric::alpha_u,alpha_v
+		float alpha_u[2];
+		float alpha_v[2];
+		//same as SAllDielectric::specReflectance
+		float specReflectance[3];
+		float dummy0;
+		//same as SAllDielectric::specTransmittance
+		float specTransmittance[3];
+		float dummy1;
+		float sigmaA[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec2 exponent;
+		vec3 specReflectance;
+		vec3 diffReflectance;
+	};
+	*/
+	struct alignas(16) SPhong
+	{
+		uint32_t bitfields;
+		float dummy0;
+		//if flag decides to use exponent texture, exponent[0..1] is bindless texture ID
+		//otherwise exponent[0] is constant exponent
+		float exponent[2];
+		//same as SAllDielectric::specReflectance
+		float specReflectance[3];
+		float dummy1;
+		//same as SAllPlastic::diffReflectance
+		float diffReflectance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec2 alpha_u;
+		vec2 alpha_v;
+		vec3 specReflectance;
+		vec3 diffReflectance;
+	};
+	*/
+	struct alignas(16) SWard
+	{
+		uint32_t bitfields;
+		float dummy0;
+		//same as for SAllDielectric::alpha_u,alpha_v
+		float alpha_u[2];
+		float alpha_v[2];
+		//same as SAllDielectric::specReflectance
+		float specReflectance[3];
+		float dummy1;
+		//same as SAllPlastic::diffReflectance
+		float diffReflectance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		uint weightsOffset;
+	};
+	*/
+	struct alignas(16) SMixture
+	{
+		//children index (into BSDF buffer) is stored on 15 highest bits
+		uint32_t bitfields;
+		//index into weights buffer
+		uint32_t weightsOffset;
+	} PACK_STRUCT;
```

Comment:
decompose a mixture BSDF into a chain of Blend BSDFs with constant (non texture) weights